File: 17solve/17-156 var2.py
# Автор В.Н. Шубинкин

mx = -20000
count = 0
pos = -1
with open('17-2.txt') as f:
    for i, line in enumerate(f.readlines(), 1):
        a = int(line)
        if a > mx:
            mx = a
            pos = i
            count = 0
        if a == mx:
            count += 1
print(count, pos)
            

File: 17solve/17-156 var2.pas
// Автор В.Н. Шубинкин
// Эффективный однопроходный алгоритм
##
var n := 10000; // количество элементов подсмотрено в текстовом файле
assign(input, '17-2.txt');
var mx := -20000;
var count := 0;
var pos := 0;
for var i := 1 to n do
begin
  var a := ReadInteger;
  if a > mx then 
    begin
      mx := a;
      pos := i;
      count := 0
    end;
  if a = mx then count += 1;
end;
print(count, pos)